src/dotty/tools/dotc/core/tasty/TreePickler.scala

```diff
@@ -438,15 +438,10 @@ class TreePickler(pickler: TastyPickler) {
       case SeqLiteral(elems, elemtpt) =>
         writeByte(REPEATED)
         withLength { pickleTree(elemtpt); elems.foreach(pickleTree) }
-      case tree: Inlined =>
-        // Why drop Inlined info when pickling?
-        // Since we never inline inside an inlined method, we know that
-        // any code that continas an Inlined tree is not inlined itself.
-        // So position information for inline expansion is no longer needed.
-        // The only reason to keep the inline info around would be to have fine-grained
-        // position information in the linker. We should come back to this
-        // point once we know more what we would do with such information.
-        pickleTree(Inliner.dropInlined(tree))
+      case Inlined(call, bindings, expansion) =>
```

Comment:
It might be possible in some situations, but it does not look straightforward. In other situations (e.g. inlined trees materialized by implicit macros) it's impossible.